sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_file_client_async.py

```diff
@@ -305,13 +355,20 @@ async def get_file_properties(self, **kwargs):
                 :dedent: 4
                 :caption: Getting the properties for a file.
         """
-        return await self._get_path_properties(cls=deserialize_file_properties, **kwargs)
+        upn = kwargs.pop('upn', None)
+        if upn:
+            headers = kwargs.pop('headers', {})
+            headers['x-ms-upn'] = str(upn)
+            kwargs['headers'] = headers
```

Comment:
Was this `upn` stuff just missed in async?

Reply:
Yes